docs/0010-differential-n-tuple-omit-prefix-storage-layout.md

```diff
@@ -0,0 +1,138 @@
+    identifier, its last occurence (right-most) will be used to select the termination of the prefix.
+  * **Type:** string
+  * **Constraints:** Must not be empty
+  * **Default:** :
+* **Name**: `tupleSegments`
+  * **Description:** Indicates the tuple segment sizes (in characters) to split the identifier into
+  * **Type:** array
+  * **Constraints:** An array of integers, each of which corresponds to a tuple segment size
+  * **Default:** [2, 3, 2, 4]
+* **Name:** `fullIdentifierAsObjectRoot`
+  * **Description:** When true, indicates that the prefix-omitted, object identifier will be included as the leaf node of the layout, holding the object root.
+  * **Type:** boolean
+  * **Default:** false
+
+## Procedure
+
+The following is an outline of the steps to map an OCFL object identifier to an OCFL object root path:
+
+1. Remove the prefix, which is everything to the left of the right-most instance of the delimiter, as well as the delimiter. If there is no delimiter, the whole id is used; if the delimiter is found at the end, an error is thrown.
+1. Starting at the leftmost character of the resulting id and working right, divide the id into `tupleSegments`. If the length of the identifier does not equal the sum of the `tupleSegments`, an error is thrown.
+1. Create the start of the object root path by joining the tuples, in order, using the filesystem path separator.
+1. Optionally, if `fullIdentifierAsObjectRoot` is true, complete the object root path by joining the prefix-omitted id (from step 1) onto the end.
+
+## Examples
+
+### Example 1
+
+This example demonstrates mappings where the single-character delimiter is found one or more times in the OCFL object
+identifier, with default `tupleSegments` and `fullIdentifierAsObjectRoot` false.
+
+#### Parameters
+
+```json
+{
+    "extensionName": "0010-differential-n-tuple-omit-prefix-storage-layout",
+    "delimiter": ":",
+    "tupleSegments": [2, 3, 2, 4],
+    "fullIdentifierAsObjectRoot": false
+}
+```
+
+#### Mappings
+
+| Object ID | Object Root Path |
+| --- | --- |
+| druid:gh875jh5489 | `gh/875/jh/5489` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.1
+├── ocfl_layout.json
+├── extensions/
+│   └── 0010-differential-n-tuple-omit-prefix-storage-layout/
+│       └── config.json
+├── gh/
+│   └── 875/
+│       └── jh/
+│           └── 5489/
+│               ├── 0=ocfl_object_1.1
+│               ├── inventory.json
+│               ├── inventory.json.sha512
+│               └── v1 [...]
+```
+
+### Example 2
+
+This example demonstrates mappings where the multi-character delimiter is found one or more times in the OCFL object identifier, with custom `tupleSegments`, a custom `delimiter`, and `fullIdentifierAsObjectRoot` turned on.
+
+#### Parameters
+
+```json
+{
+    "extensionName": "0010-differential-n-tuple-omit-prefix-storage-layout",
+    "delimiter": "edu/",
```

Comment:
This example worries me a little. The use of `edu/` as a delimiter is, in this example, a crude way to select the local path component of a full HTTP URI. One cannot do that in general with a simple string delimiter (as the second mapping example shows). If this is an intended use for the delimiter then I'm not sure that a simple delimiter string it the right thing to build into the extension. It could instead describe a general mechanism for removing the scheme & host parts of a URI, for example.

Reply:
IIRC, this is straight-up copied from 0007, and I don't consider it necessary for this extension or my organization's needs, so maybe it's best for me to remove/rework this.

Reply:
@zimeon On one hand, my use cases do not require caring about multi-character delimiters (or deconstructing URIs more narrowly). On the other, this extension is based on 0007, which apparently does. Two options we can pursue (and I'm happy to consider others):

1. Drop the multi-character delimiter from this document and note this discrepancy from 0007 above.
2. Ship it as is since this same text is already accepted in a published extension.

Have a preference? Either's fine by me.

Reply:
For now, I'm assuming 2 is fine because it lets me be lazy while I wait to hear back.

Reply:
Thanks for your patience @mjgiarlo . We discussed this in the editors' meeting and agree that the example is OK, I was perhaps overthinking it!